client/extensions/zoninator/components/data/query-zones/index.jsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { isFetchingZones } from '../../../state/zones/selectors';
+import { fetchZones } from '../../../state/zones/actions';
+
+class QueryZones extends Component {
+
+	componentWillMount() {
+		this.fetchZones( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.siteId || this.props.siteId === nextProps.siteId ) {
+			return;
+		}
+
+		this.fetchZones( nextProps );
+	}
+
+	fetchZones( props ) {
+		const { fetchingZones, siteId } = props;
+
+		if ( ! fetchingZones && siteId ) {
+			props.fetchZones( siteId );
```

Comment:
Unfortunately, no. The linter has a hard time dealing with `fetchZones` already being defined in the import above.